### PR TITLE
Infinite scroll, common tasks...

### PR DESCRIPTION
**Pull request by 1dynamite**

### CODE REVIEW

File: server/auth/is-authorized.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const createError = require("http-errors");
+
+const isAuthorized = (req, res, next) => {
+  const yes = req.user._id.toString() === req.params.userId;
```
Comment (VladimirTykvach) [inline server/auth/is-authorized.js:4]: You should validate JWT token

File: server/email.template.ejs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<p>Click the link below to confirm your email</p>
+<div>
+  <a href="http://localhost:4200/after-confirm/<%=userId%>/<%=token%>">
```
Comment (VladimirTykvach) [inline server/email.template.ejs]: localhost:4200 - shouldn't be hardcoded

File: server/controllers/users.controller.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  req.token = newToken;
+
+  next();
+};
+
+const sendLinkToEmail = async (req, res, next) => {
+  const transporter = nodemailer.createTransport({
+    service: "Gmail",
+    auth: {
+      user: "s.turaev.lab@gmail.com",
```
Comment (VladimirTykvach) [inline server/controllers/users.controller.js:72]: It should not be hardcoded

File: client/src/app/routes/dashboard/routes/tasks-route/tasks-route.component.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  templateUrl: './tasks-route.component.html',
+  styleUrls: ['./tasks-route.component.scss'],
+})
+export class TasksRouteComponent implements OnInit {
+  baseUrlTasks: string | undefined;
+
+  constructor(private router: Router) {}
+
+  ngOnInit(): void {
+    const userId = localStorage.getItem('userId');
```
Comment (VladimirTykvach) [inline client/src/app/routes/dashboard/routes/tasks-route/tasks-route.component.ts:16]: It is very weird logic, please change auth mechanism at all. You should use guard and protect some routes by it

File: server/auth/is-authorized.js (hunk 1 above)
Comment (1dynamite) [reply server/auth/is-authorized.js:4]: So, i have two middlewares named _is-signed-in_ and _is-authorized_ . _is-signed-in_ checks if the user is signed in and _is-authorized_  checks if the user is trying to CRUD his/her own data. If he/she is trying to modify some other users' data, then the request will be invalidated

Comment (VladimirTykvach) [reply server/auth/is-authorized.js:4]: Why do we need it? According to our current project we don't have any roles and policies, it means that each user can rights for CRUD (sure only for his tasks and projects)

File: client/src/app/routes/dashboard/routes/tasks-route/tasks-route.component.ts (hunk 4 above)
Comment (1dynamite) [reply client/src/app/routes/dashboard/routes/tasks-route/tasks-route.component.ts:16]: What do you mean "weird logic"? Can you explain it a bit?

File: server/auth/is-authorized.js (hunk 1 above)
Comment (1dynamite) [reply server/auth/is-authorized.js:4]: Is that a big of a deal? Should i delete it and rewrite my code?

Comment (1dynamite) [reply server/auth/is-authorized.js:4]: So, the API i wrote contains user ID. Controllers grab that user ID through _req.params.userId_ to serve requested data. According to REST principles, resource identification should be done within the API string, NOT through tokens, right? Besides, my approach is a general solution that also works for cases where there are, say, some team leaders that need data from all team members. So my code is future-proof and might save me a lot of refactoring in the future. 

So what do you think? Am i making sense?

File: client/src/app/routes/dashboard/components/tasks-rewritten/services/tasks-rewritten.service.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
```
Comment (VladimirTykvach) [inline client/src/app/routes/dashboard/components/tasks-rewritten/services/tasks-rewritten.service.ts]: I don't understand the name of file. What does "rewritten" mean?

File: server/services/tasks.service.js [hunk 6]
```diff
@@ -2,8 +2,11 @@ const Tasks = require("../models/tasks.model");
 const projectsService = require("../services/projects.service");
 const _myCounter = require("../services/_counter.service");
 
-function getTasks(filter, sort) {
-  return Tasks.find(filter).sort(sort);
+function getTasks(filter, sort, pageNumber) {
+  return Tasks.find(filter)
+    .sort(sort)
+    .skip(pageNumber > 0 ? (pageNumber - 1) * 6 : 0)
+    .limit(6);
```
Comment (VladimirTykvach) [inline server/services/tasks.service.js]: Please move it to constants

File: client/src/app/routes/dashboard/routes/project-item-route/project-item-route.component.html [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<div class="root">
+  <!-- <app-tasks [baseUrl]="baseUrlTasks"></app-tasks> -->
+  <app-tasks-rewritten
```
Comment (VladimirTykvach) [inline client/src/app/routes/dashboard/routes/project-item-route/project-item-route.component.html]: It is still strange name

**Review by VladimirTykvach — APPROVED** [review]

**Pull request merged**